Sources/_StringProcessing/RegexDSL/Core.swift

```diff
@@ -58,86 +63,109 @@ public struct Regex<Capture>: RegexProtocol {
+        let typeErasedMatch = captures.matchValue(
+          withWholeMatch: input[range]
+        )
+        convertedMatch = _openExistential(typeErasedMatch, do: bitCastToMatch)
       }
-      return RegexMatch(range: range, captures: convertedCapture)
+      return RegexMatch(range: range, match: convertedMatch)
     }
     let executor = Executor(program: regex.program.loweredProgram)
     guard let result = executor.execute(input: input) else {
       return nil
     }
-    let convertedCapture: Capture
-    if Capture.self == DynamicCaptures.self {
-      convertedCapture = DynamicCaptures.tuple([]) as! Capture
+    let convertedMatch: Match
+    if Match.self == Tuple2<Substring, DynamicCaptures>.self {
+      convertedMatch = Tuple2(
+        input[result.range], DynamicCaptures.empty
+      ) as! Match
     } else {
-      convertedCapture = () as! Capture
+      assert(Match.self == Substring.self)
+      convertedMatch = input[result.range] as! Match
     }
-    return RegexMatch(range: result.range, captures: convertedCapture)
+    return RegexMatch(range: result.range, match: convertedMatch)
   }
 }
 
 extension String {
-  public func match<R: RegexProtocol>(_ regex: R) -> RegexMatch<R.Capture>? {
+  public func match<R: RegexProtocol>(_ regex: R) -> RegexMatch<R.Match>? {
     regex.match(in: self)
   }
 
   public func match<R: RegexProtocol>(
     @RegexBuilder _ content: () -> R
-  ) -> RegexMatch<R.Capture>? {
+  ) -> RegexMatch<R.Match>? {
     match(content())
   }
 }
 
-public struct MockRegexLiteral<Capture>: RegexProtocol {
+public struct MockRegexLiteral<Match: MatchProtocol>: RegexProtocol {
   public typealias MatchValue = Substring
-  public let regex: Regex<Capture>
+  public let regex: Regex<Match>
 
   public init(
     _ string: String,
     _ syntax: SyntaxOptions = .traditional,
-    capturing: Capture.Type = Capture.self
+    matching: Match.Type = Match.self
   ) throws {
     regex = Regex(ast: try parse(string, syntax))
   }
 }
 
-public func r<C>(
-  _ s: String, capturing: C.Type = C.self
-) -> MockRegexLiteral<C> {
-  try! MockRegexLiteral(s, capturing: capturing)
+public func r<Match>(
+  _ s: String, matching matchType: Match.Type = Match.self
+) -> MockRegexLiteral<Match> {
+  try! MockRegexLiteral(s, matching: matchType)
 }
 
 fileprivate typealias DefaultEngine = TortoiseVM
 
-public protocol EmptyProtocol {}
-public struct Empty: EmptyProtocol {}
-extension Array: EmptyProtocol where Element: EmptyProtocol {}
-extension Optional: EmptyProtocol where Wrapped: EmptyProtocol {}
+public protocol EmptyCaptureProtocol {}
+public struct EmptyCapture: EmptyCaptureProtocol {}
+extension Array: EmptyCaptureProtocol where Element: EmptyCaptureProtocol {}
+extension Optional: EmptyCaptureProtocol where Wrapped: EmptyCaptureProtocol {}
 
+public protocol MatchProtocol {
+  associatedtype Capture
+}
+extension Substring: MatchProtocol {
+  public typealias Capture = EmptyCapture
+}
```

Comment:
I don't recall whether modern computing tolerates no newline at the end of a file.

Reply:
Fixed. It's partly because I've been trying to use Windows to develop these days, trying Swift for Windows myself.